accepted/2020/workloads/workload-resolvers.md

```diff
@@ -0,0 +1,237 @@
+  - Have `Success` set to true
+  - Return no SDK paths
+  - Include a `MissingWorkloadPack` item to add to the evaluation result.  The identity should be the name of the workload pack / MSBuild SDK that was requested.  It should have `Version` metadata set to the version that was read from the workload manifest.
+
+However, if the requested SDK name is `Microsoft.NET.SDK.WorkloadAutoImportPropsLocator`, then the resolver should instead behave as follows:
+
+- Look in all workload manifests for all listed workload packs
+- Find all installed workload packs from those manifests
+- From the installed packs, find all of the packs that have an `Sdk/AutoImport.props` file in them
+- Return the Sdk path for all of the packs that had the `Sdk/AutoImport.props` file in them
+
+### SDK behavior
+
+The .NET SDK (Microsoft.NET.Sdk) should:
+
+- Set the `WorkloadManifestRoot` property to a path where workload manifest packages for the current version band will be laid out.
+  - This should be `<DOTNET ROOT>/workloadmanifests/<VERSION BAND>/`
+  - For example, `c:\Program Files\dotnet\workloadmanifests\5.0.100\`
+  - Workload manifest packages should be laid out in folders under the root corresponding to the name of the workload manifest, but should not include a version in the path.  They should be updated in-place.  This allows the SDK to import all of the active `WorkloadManifest.targets` files with a simple wildcard import
+- Include the following .targets import:
+
+```xml
+<Import Project="$(WorkloadManifestRoot)/*/WorkloadManifest.targets" />
+```
+- Run target to generate appropriate build error if there are any `MissingWorkloadPack` items
+    - This should run early in the build, probably by hooking to run before the `_CheckForInvalidConfigurationAndPlatform` target
+    - The target should (via a task) look at the workload manifests to determine which workload need to be installed to supply the workload packs which are missing.  The error message generated should include this list of workloads that need to be installed.
+
+- Include the following import in one of its .props files:
+
+```xml
+<Import Project="AutoImport.props" Sdk="Microsoft.NET.SDK.WorkloadAutoImportPropsLocator">
+```
+
+### VS behavior
+
+When the VS project system loads a project, it should check the evaluation result for `MissingWorkloadPack` items.  If there are any, then it should raise the VS event to veto the project load and trigger the in-product acquisition experience for those workloads.
+
+In the same way that the SDK error generation does, the project system should map from the missing workload packs to the missing workloads via the workload manifests.  Additionally, it will need to map from the workload IDs that the .NET SDK uses to the right IDs that VS will use to install the workloads.  That mapping should either be defined in the workload manifest or in another file that is part of the same packgae.
```

Comment:
> Additionally, it will need to map from the workload IDs that the .NET SDK uses to the right IDs that VS will use to install the workloads.  That mapping should either be defined in the workload manifest or in another file that is part of the same packgae.

We should make VS Installer just another installer, and avoid making the .NET SDK and the VS Installer coupled. I think we should start with the mapping file idea from the start. If we cannot afford that in .NET 5.0, we should intend to fix that for 6.0.

```suggestion
In the same way that the SDK error generation does, the project system should map from the missing workload packs to the missing workloads via the workload manifests.  Additionally, it will need to map from the workload IDs that the .NET SDK uses to the right IDs that VS will use to install the workloads.  That mapping should either be defined in the workload manifest or in another file that is part of the same package.
```